nucypher/blockchain/eth/deployers.py

```diff
@@ -105,15 +106,25 @@ def contract(self):
     def dispatcher(self):
         return self.__proxy_contract
 
-    @property
-    def is_deployed(self) -> bool:
-        return bool(self._contract is not CONTRACT_NOT_DEPLOYED)
-
```

Comment:
Yes! I had a very similar change in a WIP PR. This property was just checking if the contract was deployed with that very same instance of the deployer, which of course is not very useful in use cases like upgrades or rollbacks. Glad this is changed :)